s4e-backend/src/main/java/pl/cyfronet/s4e/controller/GroupController.java

```diff
@@ -0,0 +1,177 @@
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Successfully retrieved a group"),
+            @ApiResponse(code = 400, message = "Group not retrieved"),
+            @ApiResponse(code = 404, message = "Group not found")
+    })
+    @GetMapping("/institutions/{institution}/groups/{group}")
+    public GroupResponse get(@PathVariable("group") String groupSlug,
+                             @PathVariable("institution") String institutionSlug)
+            throws NotFoundException, BadRequestException {
+        val group = groupService.getGroup(groupSlug)
+                .orElseThrow(() -> new NotFoundException("Group not found for id '" + groupSlug));
+        checkInstitution(institutionSlug, group);
+        return GroupResponse.of(group);
+    }
+
+    @ApiOperation("Update a group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If group was updated"),
+            @ApiResponse(code = 400, message = "Group not updated"),
+            @ApiResponse(code = 404, message = "Group was not found")
+    })
+    @PutMapping("/institutions/{institution}/groups/{group}")
+    public ResponseEntity<?> update(@RequestBody String name,
+                                    @PathVariable("group") String groupSlug,
+                                    @PathVariable("institution") String institutionSlug)
+            throws NotFoundException, GroupUpdateException, BadRequestException {
+        val group = groupService.getGroup(groupSlug)
+                .orElseThrow(() -> new NotFoundException("Group not found for id '" + groupSlug));
+        checkInstitution(institutionSlug, group);
+        group.setName(name);
+        group.setSlug(slugService.slugify(name));
+        groupService.update(group);
+        return ResponseEntity.ok().build();
+    }
+
+    @ApiOperation("Delete a group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If group was deleted"),
+            @ApiResponse(code = 400, message = "Group was deleted")
+    })
+    @DeleteMapping("/institutions/{institution}/groups/{group}")
+    public ResponseEntity<?> delete(@PathVariable("group") String groupSlug,
+                                    @PathVariable("institution") String institutionSlug)
+            throws NotFoundException, BadRequestException {
+        val group = groupService.getGroup(groupSlug)
+                .orElseThrow(() -> new NotFoundException("Group not found for id '" + groupSlug));
+        checkInstitution(institutionSlug, group);
+        val institution = institutionService.getInstitution(institutionSlug)
+                .orElseThrow(() -> new NotFoundException("Institution not found for id '" + institutionSlug));
+        groupService.delete(group, institution);
+        return ResponseEntity.ok().build();
+    }
+
+    private void checkInstitution(String institutionSlug, Group group) throws BadRequestException {
+        if (institutionSlug != group.getInstitution().getSlug()) {
```

Comment:
String comparison, safer to use equals method I guess. With null check.

Reply:
just equals, institutionSlug is pathvariable and cannot be null :)